src/components/utils/src/threads/thread_posix.cc

```diff
@@ -275,10 +278,13 @@ void Thread::join() {
 }
 
 Thread::~Thread() {
-  finalized_ = true;
-  stopped_ = true;
-  join();
-  pthread_join(handle_, NULL);
+  if (!finalized_) {
```

Comment:
@OHerasym, I do not see value of finalized_ to be changed elsewhere but here, so this if() will always be true. Probably there is some future/design extension/intention? If someone have info - please share.